doozer/doozerlib/cli/release_gen_payload.py

```diff
@@ -402,11 +412,15 @@ async def generate_assembly_issues_report(self, assembly_inspector: AssemblyInsp
         span.add_event("Checking assembly content for inconsistencies")
         self.detect_mismatched_siblings(assembly_inspector)
         assembly_build_ids: Set[int] = self.collect_assembly_build_ids(assembly_inspector)
+        # Build IDs can either be integers (Brew builds, e.g. 3403040)
+        # or strings (Konflux builds, e.g. 'ose-4-18-vsphere-problem-detector-j5n7r'
+        assembly_brew_build_ids = list(filter(lambda build_id: isinstance(build_id, int), assembly_build_ids))
```

Comment:
```suggestion
        assembly_brew_build_ids = list(filter(lambda build_id: str(build_id).isdigit(), assembly_build_ids))
```

Reply:
Do you have any particular concerns about using `is_instance`? I am reluctant to change something that was tested as working in favor of another construct, which looks fine but could yield unexpected results.

Reply:
Yeah so from what i see is a func returning a list/set consisting of possible int and string values. If the func implementation (or somewhere up the stack whatever is parsing that int value) changes and returns build_id as a string, we would still have the logic solid.

Reply:
isdigit is pretty straightforward https://docs.python.org/3/library/stdtypes.html#str.isdigit , it shouldn't ever return `False` in case of `int`.